### DISCUSSION TITLE
GraphQL Query is not intercepted in apollo-client

### DISCUSSION THREAD

**Discussion by shoNagai** [mswjs/msw#961] [Q&A]
## Environment

| Name | Version |
| ---- | ------- |
| msw  | 0.35.0   |
| apollo/client  | 3.0.2   |
| node | 14.17.1     |
| OS   | mac os big sar 11.6     |

## Request handlers

I'm using apollo client, and I'm trying to test intercepting the result of useQuery, but it's not intercepted.
The verified minimalist repository is shown below.

https://github.com/shoNagai/example-next-msw-apollo-client

When I did server.printHandlers() in jest.setup.js, the following log was output.

```
❯ npm run test

> test
> jest --config ./jest.config.js

  console.log
    [graphql] query viewer (origin: *)
      Declaration: /Users/nagaishou/gaudiy/module/example-next-msw-apollo-client/mocks/handlers.ts:4:11
```

**Comment by kettanaito** [comment]
Hey, @shoNagai. 

Your setup is correct only you're not speciying the name of the query to mock. Take a look at [your GraphQL query](https://github.com/shoNagai/example-next-msw-apollo-client/blob/2384fa14c6388a199ba65d1b9aa9fb27499f6d8c/src/index.tsx#L6-L12):

```gql
query ViewerQuery {
  viewer {
    id
    name
    status
  }
}
```

You seemed to have specified the _field name_ ("viewer") when mocking, when you should've used the _query name_ ("ViewerQuery"):

```js
// mocks/handlers.ts
import { graphql } from "msw";

export const handlers = [
  // Note the change in the line below: `viewer` -> `ViewerQuery`
  graphql.query(`ViewerQuery`, (req, res, ctx) => {
    return res(
      ctx.data({
        viewer: { id: 1, name: "John Smith", status: "cached" },
      })
    );
  }),
];
```

**Reply by shoNagai** [reply]
Thank you for your help. @kettanaito 

I think I misspecified the query here, thank you for your help!
However, I fixed it, but it still doesn't seem to be intercepted. The following is the result of running npm run test.

```

❯ npm run test

> test
> jest --config ./jest.config.js

  console.log
    [graphql] query ViewerQuery (origin: *)
      Declaration: /Users/nagaishou/gaudiy/module/example-next-msw-apollo-client/mocks/handlers.ts:4:11

      at node_modules/msw/node/lib/index.js:8012:29
          at Array.forEach (<anonymous>)

  console.log
    <body>
      <div>
        <div>
          You're signed in as
           and you're

          goto

          <a
            href="/about"
          >
            static
          </a>

          page.
        </div>
      </div>
    </body>

    /Users/nagaishou/gaudiy/module/example-next-msw-apollo-client/src/index.test.tsx:25:12
      23 |     await wait(3000);
      24 |
    > 25 |     screen.debug();
         |            ^

      at logDOM (node_modules/@testing-library/dom/dist/pretty-dom.js:89:13)


```

**Reply by kettanaito** [reply]
Hm. If it still doesn't get intercepted, I suggest following this guide we have: [Debugging uncaught requests](https://mswjs.io/docs/recipes/debugging-uncaught-requests). It's likely to lead you to the root cause.
